lib/src/cpp/domain_participant/DomainParticipant.cpp

```diff
@@ -20,314 +20,437 @@
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_default_profile(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_domain_id(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& domain_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_name(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& name)
 {
-    throw Unsupported("Unsupported");
+    // Xerces objects to manage XML elements
+    xercesc::DOMConfiguration* config = NULL;
+    xercesc::DOMDocument* doc = NULL;
+    ParseXMLErrorHandler* error_handler = NULL;
+    xercesc::DOMImplementation* implementation = NULL;
+    xercesc::DOMLSOutput* output = NULL;
+    xercesc::XercesDOMParser* parser = NULL;
+    xercesc::DOMLSSerializer* serializer = NULL;
+    xercesc::XMLFormatTarget* target = NULL;
+
+    // XML nodes and values
+    xercesc::DOMNode* profiles_node = NULL;
+    xercesc::DOMNode* participant_node = NULL;
+    xercesc::DOMNode* rtps_node = NULL;
+    xercesc::DOMNode* name_node = NULL;
+    xercesc::DOMText* name_value = NULL;
+
+    // Obtain absolute path of the given file
+    std::string absolute_xml_file = "/home/jesus/Desktop/test.xml"; // TODO update
+
+    // Try to initialize XML workspace
+    try
+    {
+        xercesc::XMLPlatformUtils::Initialize();
+    }
+    catch (const xercesc::XMLException& toCatch)
+    {
+        // Unable to initialize XML workspace
+        throw eprosima::qosprof::FileNotFound(xercesc::XMLString::transcode(toCatch.getMessage()));
+        return;
+    }
+
+    // Try to set up Xerces tools
+    if (!initialize_tools(config, doc, error_handler, implementation, output, parser, serializer, target, absolute_xml_file))
+    {
+        // Close XML workspace
+        xercesc::XMLPlatformUtils::Terminate();
+        return;
+    }
+
+    // Obtain profiles node
+    try
+    {
+        profiles_node = get_node(doc, eprosima::qosprof::tag::PROFILES);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Obtain root element
+        xercesc::DOMElement* root_element = doc->getDocumentElement();
+
+        // Add profiles
+        profiles_node = (xercesc::DOMNode*) doc->createElement(xercesc::XMLString::transcode(eprosima::qosprof::tag::PROFILES));
+        root_element->appendChild(profiles_node);
+    }
+
+    // Obtain participant node with the profile id
+   try
+    {
+        participant_node = get_node(
+            profiles_node,
+            eprosima::qosprof::tag::PARTICIPANT,
+            eprosima::qosprof::tag::PROFILE_NAME,
+            profile_id);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Create if not exist
```

Comment:
```suggestion
        // Create if not existent
```

Apply in other parts of the document.